Add documentation for BEFORE / AFTER (#2121)
neuronsimulator/nrn: docs/python/modelspec/programmatic/mechanisms/nmodl2.rst

    ``TODO``: Add description and existing example mod file

BEFORE / AFTER
~~~~~~~~~~~~~~

Description:

>     BEFORE INITIAL executes just before any INITIAL blocks of any mod file execute (but after all the FInitializeHandler(0,..) are called).

Reviewer: ```suggestion
    The `BEFORE INITIAL` block executes just before any `INITIAL` blocks of all mod files (but after all the FInitializeHandler(0,..) are called).
```

Reviewer: not sure how this renders.. shouldn't the ` be a `` - also, maybe you need to make this a bullet list, otherwise it is all connected into one paragraph.

>     AFTER INITIAL executes just after the INITIAL blocks of all mod files execute (but before all the FInitializeHandler(1, ...) are called).

Reviewer: ```suggestion
    `AFTER INITIAL` executes just after the `INITIAL` blocks of all mod files execute (but before all the FInitializeHandler(1, ...) are called).
```

>     Note that the INITIAL blocks are ordered so that mechanisms that write

Reviewer: ```suggestion

    Note that the INITIAL blocks are ordered so that mechanisms that write
```

    concentrations are after [the initialization of] ions and before mechanisms that read
    concentrations.

>         
>     But that is also the case for the order of the list of mechanisms that do INITIAL, BREAKPOINT, SOLVE, etc.

Reviewer: ```suggestion
    But that is also the case for the order of the list of mechanisms that do INITIAL, BREAKPOINT, SOLVE, etc.
```

~~~~~~

Description:

>     ``BEFORE`` is used to execute something before an event.

Reviewer: ```suggestion
    ``BEFORE`` is used to execute something before one of the following NMODL blocks:
```

>     Events can be:

Reviewer: ```suggestion
```

      - ``BREAKPOINT``
      - ``INITIAL``: will be called before the ``INITIAL`` block. But when ``v`` is already set.
      - ``STEP``
    ``TODO``: Add existing example mod file

AFTER
~~~~~

Description:

>     ``AFTER`` is used to execute sometihng after an event.
>     Events can be:

Reviewer: ```suggestion
    ``AFTER`` is used to execute code after one of the following NMODL blocks:
```
